### DISCUSSION TITLE
Keeping all photos together on the filesystem (external and internal library)

### DISCUSSION THREAD

**Discussion by buha** [immich-app/immich#6136] [Q&A]
Hi all,

So I have this 300+ GBish photo library that I have configured as external library for immich. It's structured nicely by hand in folders like:
```
2022.06 - Trip to X
2023.01 - Visit Y
```
The folder names contain info that may not be in metadata, like location, or people that I was with and is therefore information that I don't want to lose and migrating the external library to the internal one is out of the question for me.

Say this external library is at `/path/to/external` and the immich internal library path is `/path/to/internal`.

I would like all the photos to be in a single path, be it the external library and the immich managed one. Is this possible ? I assume if I would move the immich internal library to `/path/to/external/internal` this would create problems. Ideally I'd like immich to store photos in continuation of my old library, like so:
```
2022.06 - Trip to X
2023.01 - Visit Y
2024.01 (folder created by immich at the same level)
```

I'm new to immich so I don't know if and how I could achieve this.

**Comment by aisbergde** [comment]
why this should not work?

```
    volumes:
      - ${UPLOAD_LOCATION}:/usr/src/app/upload
      # added for external libraries
      - ${EXTERNAL_PATH}/photo:/usr/src/app/external/common/photo:ro
```

and just use your folders, as in your example.
